### PR TITLE
Support sofa mosn

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Motivation:

add mosn registry 

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: core-impl/client/src/main/java/com/alipay/sofa/rpc/client/router/MeshRouter.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * @author <a href="mailto:zhiyuan.lzy@antfin.com">zhiyuan.lzy</a>
+ */
+@Extension(value = "mesh", order = -19000)
+@AutoActive(consumerSide = true)
+public class MeshRouter extends Router {
+
+    /**
+     * 路由路径：注册中心
+     *
+     * @since 5.2.0
```
Comment (JervyShi) [inline core-impl/client/src/main/java/com/alipay/sofa/rpc/client/router/MeshRouter.java]: since 5.5.0

File: extension-impl/pom.xml [hunk 2]
```diff
@@ -32,6 +32,7 @@
         <module>tracer-opentracing</module>
         <module>tracer-opentracing-resteasy</module>
         <module>metrics-lookout</module>
+        <module>registry-mesh</module>
```
Comment (JervyShi) [inline extension-impl/pom.xml]: Placed under `registry-zk`

File: extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/MeshRegistry.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    /**
+     * can be extended
+     *
+     * @param config
+     * @return
+     */
+    protected ApplicationInfoRequest buildApplicationRequest(ConsumerConfig config) {
+        ApplicationInfoRequest applicationInfoRequest = new ApplicationInfoRequest();
+        applicationInfoRequest.setAppName(config.getAppName());
+        applicationInfoRequest.setAntShareCloud(true);
```
Comment (JervyShi) [inline extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/MeshRegistry.java]: setAntShareCloud should read from some config properties

File: extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/MeshConstants.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package com.alipay.sofa.rpc.registry.mesh.model;
+
+/**
+ * @author bystander
+ * @version $Id: MeshEndpoint.java, v 0.1 2018年04月20日 3:08 PM bystander Exp $
+ */
+public class MeshConstants {
+
+    public final static int TCP_PORT  = 12220;
```
Comment (JervyShi) [inline extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/MeshConstants.java]: Constant or config?

File: extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/ApplicationInfoRequest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * @version $Id: ApplicationInfoRequest.java, v 0.1 2018年04月20日 3:19 PM bystander Exp $
+ */
+public class ApplicationInfoRequest {
+
+    private boolean antShareCloud;   //true表示蚂蚁金融云共享版，其它环境为false 默认为 false
+    private String  dataCenter;      //当前应用所在的数据中心
+    private String  appName;         //应用名 必填
+    private String  zone;            //当前应用所属的zone信息
+    private String  registryEndpoint; //配置中心Endpoint地址 可空
+    private String  accessKey;       //金融云共享版环境下的AccessKey，其它环境可不传
```
Comment (JervyShi) [inline extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/ApplicationInfoRequest.java]: Useless key `accessKey` and `secretKey`.

Comment (leizhiyuan) [reply extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/ApplicationInfoRequest.java]: model has been defined

File: extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/MeshConstants.java (hunk 4 above)
Comment (leizhiyuan) [reply extension-impl/registry-mesh/src/main/java/com/alipay/sofa/rpc/registry/mesh/model/MeshConstants.java]: will not change this pr, after we change code of mesh.

**Pull request merged**